### PR TITLE
Tagged helper for StartFunction

### PR DESCRIPTION
**Pull request by turadg**
refs: #9283

## Description

https://github.com/Agoric/agoric-sdk/pull/9283 was [failing integration](https://github.com/Agoric/agoric-sdk/actions/runs/8927087379/job/24519667094?pr=9283) with,
```
Error: src/proposals/econ-behaviors.js(4,1): error TS9006: Declaration emit for this file requires using private name 'StartFunction' from module '"/home/runner/work/agoric-sdk/agoric-sdk/packages/zoe/src/zoeService/utils"'. An explicit type annotation may unblock declaration emit.
```

I've been meaning to adopt the `Tagged` helper from types-fest so I took this opportunity to try it and it solved this problem. So this PR adds it to `@agoric/internal` and uses it for the `Installation` and `Instance` types that need a tag. It doesn't use if for the three other `declare const` we have in the repo.

While debugging I also noticed we could cut a few ambient runtime imports so I did that. One required moving ManualTimer type into its impl module.

### Security Considerations
I vendored the file instead of importing from NPM to not take the external dep

### Scaling Considerations
n/a, types

### Documentation Considerations
should not be developer-facing

### Testing Considerations
CI

### Upgrade Considerations
n/a, types

### CODE REVIEW

File: packages/internal/src/tagged.d.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
+/** @file adapted from https://raw.githubusercontent.com/sindresorhus/type-fest/main/source/opaque.d.ts */
```
Comment (dckc) [inline packages/internal/src/tagged.d.ts:1]: What's the license? ah. CC0 Public Domain Dedication. very well.

might be worth noting that.

File: packages/zoe/src/zoeService/utils.d.ts [hunk 2]
```diff
@@ -13,22 +14,24 @@ type ContractFacet<T extends {} = {}> = {
...
-  getBundle: () => SourceBundle;
-  getBundleLabel: () => string;
-  // because TS is structural, without this the generic is ignored
-  [StartFunction]: SF;
-};
-export type Instance<SF> = Handle<'Instance'> & {
-  // because TS is structural, without this the generic is ignored
-  [StartFunction]: SF;
-};
+export type Installation<SF extends ContractStartFunction | unknown> = Tagged<
```
Comment (dckc) [inline packages/zoe/src/zoeService/utils.d.ts:17]: I'm not sure how to read this. `X | unknown` is `unknown` for any `X`, isn't it?

File: packages/zoe/src/zoeService/utils.d.ts [hunk 3]
```diff
@@ -13,22 +14,24 @@ type ContractFacet<T extends {} = {}> = {
 /**
  * Installation of a contract, typed by its start function.
  */
-declare const StartFunction: unique symbol;
```
Comment (dckc) [inline packages/zoe/src/zoeService/utils.d.ts:16]: ok, so no more unique symbol... though we take on responsibility for any collisions for strings such as `StartFunction`.

**Review by dckc — APPROVED** [review]
good stuff!

File: packages/zoe/src/zoeService/utils.d.ts (hunk 2 above)
Comment (turadg) [reply packages/zoe/src/zoeService/utils.d.ts:17]: this used to be `SF` which was `any`. I constrained it to `SF extends ContractStartFunction` but some places explicitly pass `unknown`. So this continues to allow that.

**Pull request merged**